pymc_bart/bart.py

```diff
@@ -109,6 +109,7 @@ def __new__(
         beta: float = 2.0,
         response: str = "constant",
         split_prior: Optional[List[float]] = None,
+        split_rules: Optional[List] = None,
```

Comment:
Shall we keep the exact expected types (maybe `ContinuousSplitRule` from what I see below)?

Reply:
This is where I need a bit of help. If you look at split_rules.py you see there are multiple different split rule options.
They essentially form an abstract class, but as they have staticmethods (for performance reasons) I don't know if it is possible to abstract them into a SplitRule class type in any reasonable way that actually enforces the static methods existing?

Reply:
Did my best here: created a class SplitRule with two static abstract methods and made other split rules inherit from it, then added SplitRule as hint here.

Reply:
ok! It seems `mypy` does not like abstract classes type=hints (https://github.com/python/mypy/issues/4717. My suggestion is to focus on the core functionalities and test and come back to this minor detail once we are ready to merge :) (From my side, I will investigate how to do it)